src/main/java/bio/overture/ego/service/RefreshContextService.java

```diff
@@ -0,0 +1,169 @@
+package bio.overture.ego.service;
+
+import static bio.overture.ego.model.enums.JavaFields.REFRESH_ID;
+import static bio.overture.ego.model.enums.StatusType.APPROVED;
+import static bio.overture.ego.model.exceptions.NotFoundException.checkNotFound;
+import static bio.overture.ego.model.exceptions.UniqueViolationException.checkUnique;
+import static java.util.Objects.isNull;
+
+import bio.overture.ego.model.domain.RefreshContext;
+import bio.overture.ego.model.entity.RefreshToken;
+import bio.overture.ego.model.entity.User;
+import bio.overture.ego.model.exceptions.ForbiddenException;
+import bio.overture.ego.repository.RefreshTokenRepository;
+import bio.overture.ego.repository.queryspecification.builder.RefreshTokenSpecificationBuilder;
+import java.sql.Date;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Optional;
+import java.util.UUID;
+import javax.servlet.http.Cookie;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+public class RefreshContextService extends AbstractBaseService<RefreshToken, UUID> {
+
+  private RefreshTokenRepository refreshTokenRepository;
+  private TokenService tokenService;
+  /** Configuration */
+  private int durationMs;
+
+  @Autowired
+  public RefreshContextService(
+      @NonNull RefreshTokenRepository refreshTokenRepository,
+      @NonNull TokenService tokenService,
+      @Value("${refreshToken.durationMs:43200000}") int durationMs) {
+    super(RefreshToken.class, refreshTokenRepository);
+    this.refreshTokenRepository = refreshTokenRepository;
+    this.tokenService = tokenService;
+    this.durationMs = durationMs;
+  }
+
+  @SuppressWarnings("unchecked")
+  public RefreshToken get(@NonNull UUID id, boolean fetchUser) {
+    val result =
+        (Optional<RefreshToken>)
+            getRepository()
+                .findOne(new RefreshTokenSpecificationBuilder().fetchUser(fetchUser).buildById(id));
+    checkNotFound(result.isPresent(), "RefreshToken '%s' does not exist", id);
+    return result.get();
+  }
+
+  @Override
+  public RefreshToken getWithRelationships(@NonNull UUID userId) {
+    return get(userId, true);
+  }
+
+  private RefreshToken createTransientToken(UUID jti) {
+    val now = Instant.now();
+    val expiry = now.plus(durationMs, ChronoUnit.MILLIS);
+    return RefreshToken.builder()
+        .jti(jti)
+        .issueDate(Date.from(now))
+        .expiryDate(Date.from(expiry))
+        .build();
+  }
+
+  public Optional<RefreshToken> getRefreshTokenByUser(User user) {
+    return refreshTokenRepository.getByUser(user);
+  }
+
+  private void checkUniqueByUser(User user) {
+    checkUnique(
+        !getRefreshTokenByUser(user).isPresent(),
```

Comment:
avoid negating during conditionals if you have access to a Boolean method that satisfies the semantics and can use it without negating. 

```java
getRefreshTokenByUser(user).isEmpty(),
```